bigquery/google/cloud/bigquery/external_config.py

```diff
@@ -0,0 +1,485 @@
+
+    def to_api_repr(self):
+        """Build an API representation of this object.
+
+        :rtype: dict
+        :returns: A dictionary in the format used by the BigQuery API.
+        """
+        config = copy.deepcopy(self._properties)
+        if self.schema is not None:
+            config['schema'] = {'fields': _build_schema_resource(self.schema)}
+        if self.options is not None:
+            config[self.options._RESOURCE_NAME] = self.options.to_api_repr()
+        return config
+
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct a CSVOptions given its API representation
+
+        :type resource: dict
+        :param resource:
+            An extract job configuration in the same representation as is
+            returned from the API.
+
+        :rtype: :class:`google.cloud.bigquery.external_config.CSVOptions`
+        :returns: Configuration parsed from ``resource``.
+        """
+        config = cls()
+        schema = resource.pop('schema', None)
+        if schema:
+            config.schema = _parse_schema_resource(schema)
+        for optcls in (BigtableOptions, CSVOptions, GoogleSheetsOptions):
+            opts = resource.pop(optcls._RESOURCE_NAME, None)
+            if opts is not None:
+                config.options = optcls.from_api_repr(opts)
+                break
+        config._properties = copy.deepcopy(resource)
+        return config
+
+
+class BigtableColumn(object):
+    """Options for a Bigtable column."""
+
+    def __init__(self):
+        self._properties = {}
+
+    encoding = _TypedApiResourceProperty(
+        'encoding', 'encoding', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.encoding
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.encoding
+    """
+
+    field_name = _TypedApiResourceProperty(
+        'field_name', 'fieldName', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.field_name
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.field_name
+    """
+
+    only_read_latest = _TypedApiResourceProperty(
+        'only_read_latest', 'onlyReadLatest', bool)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).bigtableOptions.columnFamilies.columns.only_read_latest
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.bigtableOptions.columnFamilies.columns.only_read_latest
+    """
+
+    qualifier_encoded = _TypedApiResourceProperty(
+        'qualifier_encoded', 'qualifierEncoded', six.binary_type)
```

Comment:
Does `six.binary_type` mean that we are handling the conversion to base-64 for them?

We should clarify in the docstring whether we expect binary or base-64.

Reply:
Done.